### PR TITLE
[Analytics] for onboarding events and cleanup

### PR DESCRIPTION
**Pull request by mennenia**
- Implements most of #1459 (events around prepush will come with a PR for that specific issue)
- Comments on #1459 about details around errors.

Currently contains a bug:
`Get Started` and `Login` on [this screen](https://cloud.githubusercontent.com/assets/373860/15388225/cbd83a82-1da8-11e6-8cfd-bb77507fd023.png) get triggered when the app starts, and not when the actual buttons are pressed (yet breakpoints on those selectors get triggered when expected). Am a bit baffled (can't find anything in `ARAppDelegate.m` that triggers it either) - Thoughts? cc/ @ashfurrow

### CODE REVIEW

File: Artsy/App/ARAppDelegate+Analytics.m [hunk 1]
```diff
@@ -636,88 +579,245 @@ - (void)setupAnalytics
...
-                                return controller.followedThisSession || controller.geneController.numberOfFollowedGenes;
-                            },
-                            ARAnalyticsProperties: ^NSDictionary*(ARPersonalizeViewController *controller, NSArray *_){
-                                return @{
-                                    @"artist_count": @(controller.followedThisSession),
-                                    @"gene_count": @(controller.geneController.numberOfFollowedGenes)
-                                };
-                            },
+                            ARAnalyticsEventName: ARAnalyticsOnboardingGetStarted,
+                            ARAnalyticsSelectorName: NSStringFromSelector(@selector(startOnboarding:)),
```
Comment (mennenia) [inline Artsy/App/ARAppDelegate+Analytics.m]: These are the events that are triggered upon app launch (when it loads on the home screen - I go to onboarding separately via debug menu)

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Error</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td>No CHANGELOG changes made</td>
    </tr>
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Warning</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td>Needs testing on a Phone if change is non-trivial</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
Totally forgot about this today. I'll take a look first thing tomorrow
morning.

--

[A.]           Ash Furrow

Artsy <https://www.artsy.net/>

                *ash@artsy.net <ash@artsy.net>*

                (347)-592-4474 <//(347)-592-4474> (Mobile)
                @ashfurrow <http://twitter.com/ashfurrow>

**Comment by alloy** [comment]
Going to delegate review/merging to @ashfurrow as well then.

File: Artsy/App/ARAppDelegate+Analytics.m (hunk 1 above)
Comment (ashfurrow) [inline Artsy/App/ARAppDelegate+Analytics.m]: Very strange, not sure what could be causing this. To debug, I'd recommend using a shouldFire: block, and then setting a breakpoint. Skipping over the ARAnalytics frames in the stack trace, you'll be able to see from where the method is getting called.

**Comment by ashfurrow** [comment]
Code looks good, except for the bug I suppose 😄 I'm not sure how you're doing testing for the new onboarding stuff, I didn't see any changed or added tests. If there are UI-related changes, some tests would be great. Otherwise looks 💯 and good to merge when you're comfortable @mennenia.

File: Artsy/App/ARAppDelegate+Analytics.m (hunk 1 above)
Comment (mennenia) [inline Artsy/App/ARAppDelegate+Analytics.m]: fixed

**Comment by mennenia** [comment]
Cheers! Yes tests are coming in a separate PR for the onboarding as a whole :).

**Pull request merged**